seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/redshift/RedshiftTypeMapper.java

```diff
@@ -0,0 +1,141 @@
+
+    private static final String REDSHIFT_GEOMETRY = "GEOMETRY";
+    private static final String REDSHIFT_OID = "OID";
+    private static final String REDSHIFT_SUPER = "SUPER";
+
+    private static final String REDSHIFT_TIME = "TIME";
+    private static final String REDSHIFT_TIME_WITH_TIME_ZONE = "TIME WITH TIME ZONE";
+
+    private static final String REDSHIFT_TIMETZ = "TIMETZ";
+    private static final String REDSHIFT_TIMESTAMP = "TIMESTAMP";
+    private static final String REDSHIFT_TIMESTAMP_WITH_OUT_TIME_ZONE = "TIMESTAMP WITHOUT TIME ZONE";
+
+    private static final String REDSHIFT_TIMESTAMPTZ = "TIMESTAMPTZ";
+
+    @Override
+    public SeaTunnelDataType<?> mapping(ResultSetMetaData metadata, int colIndex) throws SQLException {
+        String redshiftType = metadata.getColumnTypeName(colIndex).toUpperCase();
+        int precision = metadata.getPrecision(colIndex);
+        int scale = metadata.getScale(colIndex);
+        switch (redshiftType) {
+            case REDSHIFT_SMALLINT:
+            case REDSHIFT_INT2:
+                return BasicType.SHORT_TYPE;
+            case REDSHIFT_INTEGER:
+            case REDSHIFT_INT:
+            case REDSHIFT_INT4:
+                return BasicType.INT_TYPE;
+            case REDSHIFT_BIGINT:
+            case REDSHIFT_INT8:
+            case REDSHIFT_OID:
+                return BasicType.LONG_TYPE;
+            case REDSHIFT_DECIMAL:
+            case REDSHIFT_NUMERIC:
+                return new DecimalType(precision, scale);
+            case REDSHIFT_REAL:
+            case REDSHIFT_FLOAT4:
+                return BasicType.FLOAT_TYPE;
+            case REDSHIFT_DOUBLE_PRECISION:
+            case REDSHIFT_FLOAT8:
+            case REDSHIFT_FLOAT:
+                return BasicType.DOUBLE_TYPE;
+            case REDSHIFT_BOOLEAN:
+            case REDSHIFT_BOOL:
+                return BasicType.BOOLEAN_TYPE;
+            case REDSHIFT_CHAR:
+            case REDSHIFT_CHARACTER:
+            case REDSHIFT_NCHAR:
+            case REDSHIFT_BPCHAR:
+            case REDSHIFT_VARCHAR:
+            case REDSHIFT_CHARACTER_VARYING:
+            case REDSHIFT_NVARCHAR:
+            case REDSHIFT_TEXT:
+            case REDSHIFT_SUPER:
+                return BasicType.STRING_TYPE;
+            case REDSHIFT_DATE:
+                return LocalTimeType.LOCAL_DATE_TYPE;
+            case REDSHIFT_GEOMETRY:
+                return PrimitiveByteArrayType.INSTANCE;
+            case REDSHIFT_TIME:
+            case REDSHIFT_TIME_WITH_TIME_ZONE:
+            case REDSHIFT_TIMETZ:
+                return LocalTimeType.LOCAL_TIME_TYPE;
+            case REDSHIFT_TIMESTAMP:
+            case REDSHIFT_TIMESTAMP_WITH_OUT_TIME_ZONE:
+            case REDSHIFT_TIMESTAMPTZ:
+                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
+            default:
+                final String jdbcColumnName = metadata.getColumnName(colIndex);
+                throw new UnsupportedOperationException(
```

Comment:
Wait for https://github.com/apache/incubator-seatunnel/pull/3598 merged.May be use this is better.
```suggestion
                throw new JdbcConnectorException(CommonErrorCode.UNSUPPORTED_OPERATION,
```